### PR TITLE
Implement Go Mode

### PR DESCRIPTION
**Pull request by vicr123**
This PR implements Go Mode, allowing for bnbnav to be used as a navigational aid.

### CODE REVIEW

File: BnbnavNetClient.Linux/Program.cs [hunk 1]
```diff
@@ -2,18 +2,32 @@
...
+{
+    public static void Main(string[] args)
+    {
+        BuildAvaloniaApp().StartWithClassicDesktopLifetime(args);
+    }
+
+    static AppBuilder BuildAvaloniaApp()
+    {
+        GC.KeepAlive(typeof(SvgImageExtension).Assembly);
+        GC.KeepAlive(typeof(Avalonia.Svg.Skia.Svg).Assembly);
```
Comment (Vrabbers) [inline BnbnavNetClient.Linux/Program.cs:23]: ??

File: BnbnavNetClient/Controls/InstructionDisplayControl.axaml.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+using Avalonia;
+using Avalonia.Controls.Primitives;
+using BnbnavNetClient.Models;
+using ReactiveUI;
+
+namespace BnbnavNetClient.Controls;
+
+public class InstructionDisplayControl : TemplatedControl
+{
+    CalculatedRoute.Instruction? _Instruction;
```
Comment (Vrabbers) [inline BnbnavNetClient/Controls/InstructionDisplayControl.axaml.cs]: _instruction*

File: BnbnavNetClient/Controls/InstructionDisplayControl.axaml.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    public static readonly DirectProperty<InstructionDisplayControl, CalculatedRoute.Instruction?> InstructionProperty = AvaloniaProperty.RegisterDirect<InstructionDisplayControl, CalculatedRoute.Instruction?>("Instruction", o => o.Instruction, (o, v) => o.Instruction = v);
+    
+    int? _ToNextInstruction;
```
Comment (Vrabbers) [inline BnbnavNetClient/Controls/InstructionDisplayControl.axaml.cs]: and so on

File: BnbnavNetClient/Controls/InstructionImageControl.axaml.cs [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+using Avalonia.Media;
+using BnbnavNetClient.Helpers;
+using BnbnavNetClient.Models;
+using ReactiveUI;
+
+namespace BnbnavNetClient.Controls;
+
+public class InstructionImageControl : TemplatedControl
+{
+    CalculatedRoute.Instruction? _Instruction;
```
Comment (Vrabbers) [inline BnbnavNetClient/Controls/InstructionImageControl.axaml.cs]: ditto

File: BnbnavNetClient/Controls/InstructionImageControl.axaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+    <Style Selector="controls|InstructionImageControl">
+        <!-- Set Defaults -->
+        <Setter Property="Template">
+            <ControlTemplate>
+                
+            </ControlTemplate>
+        </Setter>
+    </Style>
+</Styles>
```
Comment (Vrabbers) [inline BnbnavNetClient/Controls/InstructionImageControl.axaml]: control with no template may be able to just have code-behind

File: BnbnavNetClient/Controls/LandmarkSearchControl.axaml.cs [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+using Avalonia.Controls.Primitives;
+using Avalonia.Data;
+using BnbnavNetClient.Models;
+using BnbnavNetClient.Services;
+
+namespace BnbnavNetClient.Controls;
+
+public class LandmarkSearchControl : TemplatedControl
+{
+    ISearchable? _SelectedLandmark;
```
Comment (Vrabbers) [inline BnbnavNetClient/Controls/LandmarkSearchControl.axaml.cs]: _selectedLandmark

File: BnbnavNetClient/Helpers/DrawingContextExtensions.cs [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+            rect.Y * sourceSize.Height / rect.Height);
+        var rotateMatrix = Matrix.CreateRotation(MathHelper.ToRad(angle));
+        var preRotateMatrix = Matrix.CreateTranslation(-sourceSize.Width / 2, -sourceSize.Width / 2);
+
+        using (context.PushClip(rect))
+        using (context.PushPreTransform(translateMatrix * scaleMatrix))
+        using (context.PushTransformContainer())
+        using (context.PushPostTransform(preRotateMatrix))
+        using (context.PushPostTransform(rotateMatrix))
+        using (context.PushPostTransform(preRotateMatrix.Invert()))
```
Comment (Vrabbers) [inline BnbnavNetClient/Helpers/DrawingContextExtensions.cs]: multiply these?

**Review by Vrabbers — CHANGES_REQUESTED** [review]

File: BnbnavNetClient.Linux/Program.cs (hunk 1 above)
Comment (vicr123) [reply BnbnavNetClient.Linux/Program.cs:23]: Technical changes required for the SVG package to work - you can read more about it in the SVG.Skia package

**Review by Vrabbers — APPROVED** [review]

File: BnbnavNetClient.Mac/TextToSpeech/MacTextToSpeechProvider.cs [hunk 8]
```diff
@@ -23,7 +23,14 @@ public async Task SpeakAsync(string text)
         {
             DispatchQueue.MainQueue.DispatchSync(() =>
             {
-                var synth = new NSSpeechSynthesizer(NSSpeechSynthesizer.AvailableVoices[0]);
+                var availableVoices = NSSpeechSynthesizer.AvailableVoices.Where(x =>
```
Comment (Royce551) [inline BnbnavNetClient.Mac/TextToSpeech/MacTextToSpeechProvider.cs]: voicesMatchingCurrentCulture? or something else that's more descriptive?

File: BnbnavNetClient/App.axaml [hunk 9]
```diff
@@ -48,6 +49,41 @@
     <Style Selector="Button.destructive:pressed">
```
Comment (Royce551) [inline BnbnavNetClient/App.axaml:49]: IMO, this whole thing should be a part of DayTheme.axaml and NightTheme.axaml

**Review by Royce551 — CHANGES_REQUESTED** [review]

File: BnbnavNetClient/App.axaml (hunk 9 above)
Comment (vicr123) [reply BnbnavNetClient/App.axaml:49]: This can be resolved later because it doesn't work anyway

**Pull request merged**